WcaOnRails/app/views/competitions/_index_form.html.erb

```diff
@@ -95,4 +122,31 @@
   });
 
   $('#display-<%= @display %>').button('toggle');
+
+  // From http://stackoverflow.com/questions/901115/how-can-i-get-query-string-values-in-javascript/901144#901144
+  function getParameterByName(name, url) {
+    if (!url) url = window.location.href;
+    name = name.replace(/[\[\]]/g, "\\$&");
+    var regex = new RegExp("[?&]" + name + "(=([^&#]*)|&|#|$)"),
+        results = regex.exec(url);
+    if (!results) return null;
+    if (!results[2]) return '';
+    return decodeURIComponent(results[2].replace(/\+/g, " "));
+  }
+
+  function display_selectors() {
+    $(".admin-selectors, .map-selectors").hide();
+    $display_class = "." + getParameterByName('display') + '-selectors';
```

Comment:
This looks like an opening for an injection attack. Don't bother fixing it, however, as I have another approach I'd like you to investigate.